//
// Created by Shrekulka on 28.06.2023.
//
#include "pch.h"

#ifndef TCP_SERVER_NETWORKEXCEPTION_H
#define TCP_SERVER_NETWORKEXCEPTION_H

/*
Класс NetworkException используется для представления исключений, связанных с сетевыми операциями. Он является
наследником класса std::runtime_error, что позволяет использовать его функциональность для создания исключений с
сообщением об ошибке.

В данном коде класс NetworkException определяет конструктор, который принимает сообщение об ошибке в качестве параметра.
Внутри конструктора используется конструктор родительского класса std::runtime_error, который принимает полное сообщение
об ошибке, состоящее из переданного сообщения и сообщения об ошибке, полученного с помощью std::strerror(errno). Таким
образом, создается объект исключения с заданным сообщением и ошибкой.

Этот класс может быть полезен, если требуется выбросить исключение при возникновении сетевых ошибок и передать
дополнительное сообщение, объясняющее причину ошибки. При обработке исключения можно получить доступ к сообщению об
ошибке и ее описанию для дальнейшей обработки или вывода пользователю.
*/

class NetworkException : public std::runtime_error
{
public:
	NetworkException(std::string message) : std::runtime_error(
			message + ": " + std::strerror(errno)) // Создать исключение с сообщением и ошибкой
	{
	}
};

#endif //TCP_SERVER_NETWORKEXCEPTION_H
